src/com/nutrons/framework/commands/ParallelCommand.java

```diff
@@ -0,0 +1,64 @@
+package com.nutrons.framework.commands;
+
+import io.reactivex.Flowable;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.schedulers.Schedulers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParallelCommand implements CommandWorkUnit {
+  private final Flowable<CommandWorkUnit> commands;
+
+  ParallelCommand(CommandWorkUnit... commands) {
+    this.commands = Flowable.fromArray(commands);
+
+  }
+
+  @Override
+  public Flowable<Terminator> execute() {
+    PublishProcessor<Terminator> finisher = PublishProcessor.create();
+    final ExecuteLock lock = new ExecuteLock();
```

Comment:
Don't roll your own lock.  I think there are some in Guava that we can import.  We need to drop it to the robot because of WPILib, so if we use it too, its not a big deal